### PR TITLE
Cart page: created detail products Cart page

### PR DESCRIPTION
**Pull request by AndreeaSburlea**
## New features
- detail cart products page
- delete button for products

## Changes in code
- customized the cart page
- added a footer for the cart total
- replaced the async blocks

## Screenshot

Screenshot 2022-06-22 at 17 01 00
Screenshot 2022-06-22 at 17 01 09

### CODE REVIEW

File: ShopApp/ShopApp/Model/CartProduct.swift [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        self.size = size
+        self.price = price
+        self.quantity = quantity
+    }
+
+    init() {
+        self.init(image: UIImage(), name: "", size: "", price: 0, quantity: 0)
+    }
+}
+
```
Comment (madalinaCojocaru1) [inline ShopApp/ShopApp/Model/CartProduct.swift]: you can remove this empty line

File: ShopApp/ShopApp/View/CartProductTableViewCell.swift [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        self.productPrice.text = String(format: "%.2f", price)
+        self.productQuantity.text = String(quantity)
+        if quantity == 0{
+            self.deleteButton.isHidden = false
+        } else {
+            self.deleteButton.isHidden = true
+        }
+        self.minusButton.setTitle("-", for: .normal)
+        self.plusButton.setTitle("+", for: .normal)
+
```
Comment (madalinaCojocaru1) [inline ShopApp/ShopApp/View/CartProductTableViewCell.swift]: this one also can be removed

**Review by madalinaCojocaru1 — APPROVED** [review]

File: ShopApp/ShopApp/Controller/CartTableViewController.swift [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+        var currentQuantity = cartProducts[indexPath.row].getQuantity()
+
+        if sender.currentTitle! == "+" {
+            currentQuantity += 1
+        } else {
+            if currentQuantity > 0 {
+            currentQuantity -= 1
+            }
+        }
+        cartProducts[indexPath.row].setQuantity(quantity: currentQuantity)
```
Comment (VivianaSara) [inline ShopApp/ShopApp/Controller/CartTableViewController.swift]: here should be an empty line

File: ShopApp/ShopApp/Controller/CartTableViewController.swift [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+
+    @IBAction private func deleteCartProduct(_ sender: UIButton) {
+        guard let email = Auth.auth().currentUser?.email as? String else { return }
+        let buttonPosition: CGPoint = sender.convert(CGPoint.zero, to: self.tableView)
+        guard let indexPath1 = self.tableView.indexPathForRow(at: buttonPosition) else { return }
+         let key = ref.child("users/" + email.replacingOccurrences(of: ".", with: " ") + "/cart/" + cartProducts[indexPath1.row].getName())
+        key.removeValue { error, _ in
+            print(error as Any)
+        }
+        cartProducts.remove(at: indexPath1.row)
```
Comment (VivianaSara) [inline ShopApp/ShopApp/Controller/CartTableViewController.swift]: empty line needed

File: ShopApp/ShopApp/Controller/CartTableViewController.swift [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+    // MARK: - ViewDidLoad
+    override func viewDidLoad() {
+            super.viewDidLoad()
+            self.getAllCartProducts {
+                self.tableView.dataSource = self.dataSource
+                self.updateSnapshot()
+                self.addFooter()
+                self.getTotal()
+        }
+    }
```
Comment (VivianaSara) [inline ShopApp/ShopApp/Controller/CartTableViewController.swift]: this should be aligned

File: ShopApp/ShopApp/Controller/CartTableViewController.swift [hunk 6]
```diff
@@ -0,0 +1,202 @@
...
+
+    // MARK: - Get the cart total
+    func getTotal() {
+        var total: Float = 0
+        for cartProduct in cartProducts {
+            total += cartProduct.getPrice() * Float(cartProduct.getQuantity())
+        }
+        totalPriceLabel.text = String(format: "%.2f $", total)
```
Comment (VivianaSara) [inline ShopApp/ShopApp/Controller/CartTableViewController.swift]: empty line

File: ShopApp/ShopApp/Controller/CartTableViewController.swift [hunk 7]
```diff
@@ -0,0 +1,202 @@
...
+
+    // MARK: - Update the cart quantity in database
+    func updateQuantity(index: Int, quantity: Int) {
+        guard let email = Auth.auth().currentUser?.email as? String else { return }
+        // Get cart product key
+        guard let key = ref.child("users/" + email.replacingOccurrences(of: ".", with: " ") + "/cart/" + cartProducts[index].getName()).key else { return }
+        // Create new properties for cart product
+        let properties = ["quantity": String(quantity),
+                          "size": cartProducts[index].getSize()]
+        // Create update command
```
Comment (VivianaSara) [inline ShopApp/ShopApp/Controller/CartTableViewController.swift]: maybe you should put an empty lin before every comments

File: ShopApp/ShopApp/Controller/CartTableViewController.swift [hunk 8]
```diff
@@ -0,0 +1,218 @@
...
+
+                    // Add cart product to the list
+                    self.cartProducts.append(cartProduct)
+                } else {
+                    return
+                }
+            }
+                self.cartProducts.sort(by: { $0.getName() < $1.getName() })
+                completion()
+            })
```
Comment (VivianaSara) [inline ShopApp/ShopApp/Controller/CartTableViewController.swift]: this should be aligned and empty space before self.cart....

**Review by VivianaSara — APPROVED** [review]

**Pull request merged**